rubrics/tool/src/main/java/org/sakaiproject/rubrics/controller/GeneratePdfController.java

```diff
@@ -0,0 +1,268 @@
+import org.springframework.http.ContentDisposition;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.method.annotation.StreamingResponseBody;
+
+import com.itextpdf.text.BaseColor;
+import com.itextpdf.text.Chunk;
+import com.itextpdf.text.Document;
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.Element;
+import com.itextpdf.text.Font;
+import com.itextpdf.text.PageSize;
+import com.itextpdf.text.Paragraph;
+import com.itextpdf.text.Phrase;
+import com.itextpdf.text.pdf.PdfPCell;
+import com.itextpdf.text.pdf.PdfPTable;
+import com.itextpdf.text.pdf.PdfWriter;
+
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.PreferencesService;
+
+import org.jsoup.Jsoup;
+
+@Slf4j
+@RepositoryRestController
+@NoArgsConstructor
+@AllArgsConstructor
+public class GeneratePdfController {
+
+    @Autowired
+    RubricsConfiguration rubricsConfiguration;
+
+    @Autowired
+    RepositoryEntityLinks repositoryEntityLinks;
+    
+    @Autowired
+    private MessageSource messageSource;
+    
+    @Autowired
+    private SiteService siteService;
+    
+    @Autowired
+    private RubricRepository rubricRepository;
+    
+    @Autowired
+    private EvaluationRepository evaluationRepository;
+    
+    @Autowired
+    private SessionManager sessionManager;
+
+    @Autowired
+    private PreferencesService preferencesService;
+
+    private static final String X_COPY_HEADER = "x-copy-source";
+    private static final String X_LANG_HEADER = "lang";
+    private static Font font = new Font(Font.FontFamily.HELVETICA, 10, Font.BOLD);
+    private static Font textFont = new Font(Font.FontFamily.HELVETICA, 8, Font.BOLD);
+    private static Font descFont = new Font(Font.FontFamily.HELVETICA, 7, Font.NORMAL);
+
+    @PreAuthorize("canCopy(#sourceId, 'Rubric')")
+    @RequestMapping(value = "/rubrics/generatePdf", headers = {X_COPY_HEADER,X_LANG_HEADER}) 
+    public ResponseEntity<StreamingResponseBody> generatePdf(@Param("sourceId") @RequestHeader(value = X_COPY_HEADER) String sourceId, 
+            @RequestHeader(value = X_LANG_HEADER) String lang, @RequestHeader(value = "toolId") String toolId,
+            @RequestHeader(value = "itemId") String entityId,
+            @RequestHeader(value = "evaluatedItemId") String evaluatedItemId)
+            throws CloneNotSupportedException {
+
+        Rubric sourceRubric = null;
+        String fileRubricName = null;
+        List<Evaluation> evals = null;
+        StreamingResponseBody stream = null;
+
+        try {
```

Comment:
Okay, lets try to split the logic using try-with-resources, you're requesting rubrics entities handled by the same try catch block, try to split that logic from the I/O operations with files.

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
^